### PR TITLE
Add changes for SPV_EXT_shader_atomic_float_add

### PR DESCRIPTION
**Pull request by vkushwaha-nv**
WIP has been removed, ready to merge.

### CODE REVIEW

**Comment by vkushwaha-nv** [comment]
I see 2 failures:
1. introduction of "+Validation failed" , I can not repro this while running the failing test.
2. The log of new tests added, have a space at the end, which I am assuming git removes on commit. I tried fixing that from Parsecontextbase.cpp, however that seems to introduce unrelated failures in other tests. 

@ben-clayton  can you take a look?

**Comment by dgkoch** [comment]
> introduction of "+Validation failed" , I can not repro this while running the failing test.

This is likely because the version of spirv-tools glslang uses needs to be updated (can be done now that the spirv-tools PR was merged this morning).

**Comment by vkushwaha-nv** [comment]
Thanks Daniel. Fixed.

**Comment by johnkslang** [comment]
The spirv.hpp has conflicts with the official one from SPIRV-Headers. I've placed a copy of the official one from SPIRV-Headers into glslang now, and you should be able to drop that file from your submission.

**Comment by johnkslang** [comment]
Git should never remove any spaces, unless your edits removed spaces.  What log?  A test result?

**Comment by johnkslang** [comment]
Also, remove the WIP when you believe you're done from your side.

**Comment by vkushwaha-nv** [comment]
> Also, remove the WIP when you believe you're done from your side.

Removed. I think this is now done. Thanks!

File: SPIRV/spirv.hpp [hunk 1]
```diff
@@ -1818,6 +1818,7 @@ inline void HasResultAndType(Op opcode, bool *hasResult, bool *hasResultType) {
     case OpAtomicAnd: *hasResult = true; *hasResultType = true; break;
     case OpAtomicOr: *hasResult = true; *hasResultType = true; break;
     case OpAtomicXor: *hasResult = true; *hasResultType = true; break;
+    case OpAtomicFAddEXT: *hasResult = true; *hasResultType = true; break;
```
Comment (johnkslang) [inline SPIRV/spirv.hpp:1821]: This change should not be made to glslang's copy of the SPIRV-Headers' file.

**Comment by johnkslang** [comment]
There should not be any changes to spirv.hpp.  It is copied here from SPIRV-Headers, and is essentially considered correct by definition.

**Comment by vkushwaha-nv** [comment]
> There should not be any changes to spirv.hpp. It is copied here from SPIRV-Headers, and is essentially considered correct by definition.
Fixed, sorry it skipped my mind.

**Pull request merged**